cell_classification/segmentation_data_prep_test.py

```diff
@@ -33,9 +49,165 @@ def test_get_image():
+            dataset="dataset",
+            tile_size=[256, 256],
+            tf_record_path=os.path.join(temp_dir, "tf_record_path"),
+            normalization_dict_path=os.path.join(temp_dir, "norm_dict_test.json"),
+        )
+        data_prep.make_tf_record(
+            ["list", "of", "data", "folders"], os.path.join(temp_dir, "tf_record_path")
+        )
+        assert norm_dict == data_prep.normalization_dict
+
+        # check if the normalization_dict works correctly when
+        # data_folders but no normalization_dict_path is given to make_tf_record
+        data_prep = SegmentationTFRecords(
+            data_folders=data_folders,
+            cell_table_path="list_to_cell_table_path",
+            conversion_matrix_path=conversion_matrix_path,
+            imaging_platform="imaging_platform",
+            dataset="dataset",
+            tile_size=[256, 256],
+            tf_record_path=os.path.join(temp_dir, "tf_record_path"),
+        )
+        data_prep.make_tf_record(
+            ["list", "of", "data", "folders"], os.path.join(temp_dir, "tf_record_path")
+        )
+        assert norm_dict == data_prep.normalization_dict
+        norm_dict_loaded = json.load(
+            open(os.path.join(temp_dir, "tf_record_path", "normalization_dict.json"))
+        )
+
+        # check if the normalization_dict is correctly written to the json file
+        assert norm_dict_loaded == norm_dict
+
+
+def test_instance_mask():
+    instance_mask = np.zeros([256, 256], dtype=np.uint16)
+    instance_mask[0:32, 0:32] = 1
+    instance_mask[0:32, 32:64] = 2
+    instance_mask[0:32, 64:96] = 3
+    instance_mask[32:64, 0:32] = 4
+    instance_mask[64:96, 64:96] = 5
+
+    instance_mask_eroded = np.zeros([256, 256], dtype=np.uint8)
+    instance_mask_eroded[0:31, 0:31] = 1
+    instance_mask_eroded[0:31, 33:63] = 1
+    instance_mask_eroded[0:31, 65:95] = 1
+    instance_mask_eroded[33:63, 0:31] = 1
+    instance_mask_eroded[65:95, 65:95] = 1
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        # check if the instance_mask is correctly loaded
+        imwrite(os.path.join(temp_dir, "cell_segmentation.tiff"), instance_mask)
+        data_prep = prep_object()
+        loaded_binary_img, loaded_img = data_prep.get_instance_mask(
+            data_folder=temp_dir, segmentation_fname="cell_segmentation"
+        )
+        assert np.array_equal(loaded_img, instance_mask)
+
+        # check if binary mask is binarized correctly
+        assert np.array_equal(np.unique(loaded_binary_img), np.array([0, 1]))
+
+        # check if binary mask is eroded correctly
+        assert np.array_equal(loaded_binary_img, instance_mask_eroded)
+
+
 def test_prepare_example():
     data_prep = prep_object()
     with tempfile.TemporaryDirectory() as temp_dir:
+        data_folders = prepare_test_data_folders(5, temp_dir, selected_markers=["CD8"])
+        norm_dict = data_prep.calculate_normalization_matrix(
```

Comment:
Same thing here. We don't need to run the actual `calculate_normalization_matrix` function for this integration test. We just want any placeholder normalization dict to be attached to `data_prep`. In fact, it's often advantageous to make a placeholder normalization_dict ourselves in the test function, because then it's very obvious what the values are.